libraries/stdlib/samples/test/samples/properties/delegates.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package samples.properties
+
+import kotlin.properties.Delegates
+import samples.*
+import kotlin.test.*
+
+class Delegates {
+    @Sample
+    fun vetoableDelegate() {
+        var max: Int by Delegates.vetoable(0) { property, oldValue, newValue ->
+            newValue > oldValue
+        }
+
+        assertPrints(max, "0")
+
+        max = 10
+        assertPrints(max, "10")
+
+        max = 5
+        assertPrints(max, "10")
+    }
+
+    @Sample
+    fun throwVetoableDelegate() {
+        var max: Int by Delegates.vetoable(0) { property, oldValue, newValue ->
+            if (newValue > oldValue) true else throw(Throwable("New value must be larger than old value."))
```

Comment:
Usually `IllegalArgumentException` is used in this case. Also there are redundant parentheses here.